draft-ietf-quic-transport.md

```diff
@@ -1477,35 +1481,236 @@ connection if the integrity check fails with a PROTOCOL_VIOLATION error code.
+PATH_CHALLENGE, PATH_RESPONSE, or PADDING; and
+
+* the packet carries a packet number that is the largest seen thus far.
+
+Upon receiving such a packet, the server MUST abandon any path validation it is
+performing with other addresses on the expectation that those validations are
+likely to fail.  Abandoning path validation primarily means ceasing subsequent
+transmissions of the PATH_CHALLENGE frame.  An endpoint MUST ignore any
+subsequently received PATH_RESPONSE frames from abandoned addresses.
+
+In response to such a packet, the server MUST start sending subsequent packets
+to this client address and MUST initiate path validation to verify client
+ownership of the unvalidated address.
+
+A server MAY skip validation of a client address that it considers to be already
+valid.
+
+Note that the server MAY send data to an unvalidated client address, but it MUST
+protect against potential attacks as described in {{address-spoofing}} and
+{{on-path-spoofing}}.
+
+If a non-probing packet is received from a new client address but with a packet
+number that is not the largest seen thus far, the server must process the packet
+as usual, but it MUST ignore this client address under the assumption that it is
+not the client's most recent address.
+
+After verifying a new client address, the server SHOULD update any address
+validation tokens ({{address-validation}}) that it has issued to the client if
+those are no longer valid based on the changed address.
 
-An endpoint that receives packets that contain a source IP address and port that
-has not yet been used can start sending new packets with those as a destination
-IP address and port.  Packets exchanged between endpoints can then follow the
-new path.
 
-Due to variations in path latency or packet reordering, packets from different
-source addresses might be reordered.  The packet with the highest packet number
-MUST be used to determine which path to use.  Endpoints also need to be prepared
-to receive packets from an older source address.
+### Handling Address Spoofing by a Client {#address-spoofing}
 
-An endpoint MUST validate that its peer can receive packets at the new address
-before sending any significant quantity of data to that address, or it risks
-being used for denial of service.  See {{migrate-validate}} for details.
+It is possible that the client is spoofing its source address to cause the
+server to send excessive amounts of data to an unwilling host.  If the server
+sends significantly more data than the client, connection migration might be
+used to amplify the volume of data that an attacker can generate toward a
+victim.
+
+Thus, on receiving and decrypting a packet from an unvalidated address, the
+server MUST immediately validate the client's new address to confirm the
+client's possession of the new address.
+
+Until a client's address is deemed valid, the server MUST limit the rate at
+which it sends data to this address.  The server MUST NOT send more than 4 * MSS
```

Comment:
It's a bit arbitrary, but I was shooting for something > min cwnd. Honestly, any number SGTM. Do you prefer min cwnd? 

I assumed the initial RTO period implied a default if path RTT was unknown. Happy to clarify if you think it's not clear.

Reply:
I see why you chose 4*MSS.  In this document, 2*Min CWND seems preferable to a specific value to me.

I think it's worth clarifying that the RTO is likely based on default RTT/default RTO in this case, since if you haven't received a PATH_RESPONSE, I would expect you either have no estimate or a somewhat unverified estimate of RTT.

Reply:
Everything that Ian said, except that I don't understand why 2xMinCWND.  Seems equally arbitrary.  MinCWND on the assumption of a default RT[TO] would be more defensible.

Reply:
added text and set the rate to min cwnd per rto.